File: server/plugin/dict/router/entry.go
package router

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

var Entry = new(entry)

type entry struct{}

func (r *entry) Init(public *gin.RouterGroup, private *gin.RouterGroup) {
	{
		group := private.Group("entry").Use(middleware.OperationRecord())
		group.POST("createEntry", apiEntry.CreateEntry)
		group.DELETE("deleteEntry", apiEntry.DeleteEntry)
		group.DELETE("deleteEntryByIds", apiEntry.DeleteEntryByIds)
		group.PUT("updateEntry", apiEntry.UpdateEntry)
		group.POST("importExcel", apiEntry.ImportExcel)
	}
	{
		group := private.Group("entry")
		group.GET("findEntry", apiEntry.FindEntry)
		group.GET("getEntryList", apiEntry.GetEntryList)
		group.GET("loadDictionary", apiEntry.LoadDictionary)
	}
}

File: web/src/plugin/dict/api/entry.js
import service from '@/utils/request'

export const createEntry = (data) => {
  return service({
    url: '/entry/createEntry',
    method: 'post',
    data
  })
}

export const deleteEntry = (params) => {
  return service({
    url: '/entry/deleteEntry',
    method: 'delete',
    params
  })
}

export const deleteEntryByIds = (params) => {
  return service({
    url: '/entry/deleteEntryByIds',
    method: 'delete',
    params
  })
}

export const updateEntry = (data) => {
  return service({
    url: '/entry/updateEntry',
    method: 'put',
    data
  })
}

export const findEntry = (params) => {
  return service({
    url: '/entry/findEntry',
    method: 'get',
    params
  })
}

export const getEntryList = (params) => {
  return service({
    url: '/entry/getEntryList',
    method: 'get',
    params
  })
}

export const importExcel = (data) => {
  return service({
    url: '/entry/importExcel',
    method: 'post',
    data,
    headers: {
      'Content-Type': 'multipart/form-data'
    }
  })
}

export const loadDictionary = (params) => {
  return service({
    url: '/entry/loadDictionary',
    method: 'get',
    params
  })
}
